reference/toolstacks/php/drupal/redis-drupal-8.md

```diff
@@ -0,0 +1,154 @@
+example below is intended as a "most common case".
+
+```php
+// Set redis configuration.
+if (!empty($relationships['redis'][0]) && extension_loaded('redis')) {
+  $redis = $relationships['redis'][0];
+
+  $settings['cache']['default'] = 'cache.backend.redis';
+  $settings['redis.connection']['host'] = $redis['host'];
+  $settings['redis.connection']['port'] = $redis['port'];
+
+  // Enable the redis server overrides, e.g. cache tags checksum.
+  // See that file, you might want to copy and customize it, it is meant as an example
+  // and will change as more backends are offered by the redis module.
+  $settings['container_yamls'][] = 'modules/contrib/redis/example.services.yml';
+
+  // Manually add the classloader path, this is required for the container cache bin definition below
+  // and allows to use it without the redis module being enabled
+  $class_loader->addPsr4('Drupal\\redis\\', 'modules/contrib/redis/src');
+
+  // Use redis for container cache.
+  $settings['bootstrap_container_definition'] = [
+    'parameters' => [],
+    'services' => [
+      'redis.factory' => [
+        'class' => 'Drupal\redis\ClientFactory',
+      ],
+      'cache.backend.redis' => [
+        'class' => 'Drupal\redis\Cache\CacheBackendFactory',
+        'arguments' => ['@redis.factory', '@cache_tags_provider.container'],
+      ],
+      'cache.container' => [
+        'class' => '\Drupal\redis\Cache\PhpRedis',
+        'factory' => ['@cache.backend.redis', 'get'],
+        'arguments' => ['container'],
+      ],
+      'cache_tags_provider.container' => [
+        'class' => 'Drupal\redis\Cache\RedisCacheTagsChecksum',
+        'arguments' => ['@redis.factory'],
+      ],
+    ],
+  ];
+
+  // Explicitly set the fast backend for bootstrap, default, discover and config,
+  // otherwise this gets lost when redis is enabled.
+  $settings['cache']['bins']['bootstrap'] = 'cache.backend.chainedfast';
+  $settings['cache']['bins']['discovery'] = 'cache.backend.chainedfast';
+  $settings['cache']['bins']['config'] = 'cache.backend.chainedfast';
+  // Default does not use the fast chained backend by default, but core only uses it for small caches
+  // ensure this is also the case for a specific site before enabling this.
+  // $settings['cache']['bins']['default'] = 'cache.backend.chainedfast';
+
+  // Set a fixed prefix so that all requests share that, the default is currently not reliable.
+  $settings['cache_prefix'] = 'prefix_';
+}
+```
+
+The `example.services.yml` file noted above will also use Redis for the lock and flood
+control systems.
+
+Although the redis module is able to use Redis for a queue backend, we do not recommend
+that on Platform.sh. Our Redis instance is not persistent, which means every deploy
+would result in an emptied queue and possible lost items when the Redis instance
+gets full. It is, however, safe for transient information like caching and locking.
```

Comment:
Hm. At least the non-persistent locks unlock themself at the end of the request anyway, so I don't see a problem there. The persistent lock could be a problem, that's e.g. used for config import, but I doubt you could do something like that successfully during a re-deploy anyway? It is however possible that sites have custom uses that do need to persist for a longer time. I already mention in my comment that sites should review and possibly customize that file (by copying it and changing it), you could mention the persistent lock as an example for that.